packages/build/src/log/messages/compatibility.js

```diff
@@ -90,16 +91,19 @@ const getVersionField = function ([versionFieldName, version]) {
  * This can only happen when they are installed to `package.json`.
  *
  * @param {BufferedLogs}} logs
- * @param {object[]} outdatedPlugins
+ * @param {object[]} pluginsOptions
+ * @param {object[]} featureFlags
  *
  * @throws Error Throws an error if the Next runtime is >= 4.0.0 || < 4.26.0
  */
-export const logOutdatedPlugins = function (logs, outdatedPlugins) {
+export const logOutdatedPlugins = function (logs, pluginsOptions, featureFlags) {
+  const outdatedPlugins = pluginsOptions.filter(hasOutdatedVersion).map(getOutdatedPlugin)
   if (outdatedPlugins.length === 0) {
     return
   }
 
-  throwIfOutdatedNextRuntime(outdatedPlugins)
+  if (featureFlags.plugins_break_builds_with_unsupported_plugin_versions)
+    throwIfOutdatedNextRuntime(pluginsOptions.filter(hasOutdatedVersion))
```

Comment:
[dust] Even though it is Next specific, @ascorbic had suggested a more generic name. See https://github.com/netlify/build/pull/4668#discussion_r1012682975

Reply:
done, pushing tests in a bit